ape_hardhat/provider.py

```diff
@@ -283,11 +283,16 @@ def project_folder(self) -> Path:
 
     @property
     def uri(self) -> str:
-        if self.config.host:
-            if "https" not in self.config.host or "http" not in self.config.host:
-                self.config.host = "http://" + self.config.host  # type: ignore
         if self._host is None:
-            self._host = self.config.host or f"http://127.0.0.1:{DEFAULT_PORT}"
+            if self.config.host:
+                if "127.0.0.1" in self.config.host and "http" not in self.config.host:
```

Comment:
```suggestion
                if "127.0.0.1" in self.config.host and "http" not in self.config.host:
```
```suggestion
                if "127.0.0.1" in self.config.host or "localhost" in self.config.host and not self.config.host.startswith("http"):
```
and similar changes for the checks below